### ISSUE TITLE
Duplication bug

### ISSUE THREAD

**Issue by RobertCody16** [anegostudios/VintageStory-Issues#462]
**Game Version:** 1.12.6
**Platform:** Windows
**Modded:** No

### Description
**Note this happened while playing on a server.
I duplicated a barrel of Lime Slack using middle click and the game crashed. I also crash immediately upon loading back into the server.
The barrel started as 10 Litters of water and I added 40 QuickLime.
Duplicated with middle click, tried to Shift place the new barrel, and the game crashed.
### How to reproduce
1. Could not reproduce in a single player instance.
2. This happened on a server.
3. Log Linked below

### Expected behavior
To duplicate no problem

### Screenshots
Attach a screenshot if it helps explain the issue / shows a visual bug.

### Logs
```
Paste crash here.

Running on 64 bit Windows with 12 GB RAM 
Version: v1.12.6 (Stable)3/9/2020 1:16:58 AM: Critical error occurred
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Vintagestory.GameContent.BlockBarrel.OnBeforeRender(ICoreClientAPI capi, ItemStack itemstack, EnumItemRenderTarget target, ItemRenderInfo& renderinfo)
   at Vintagestory.Client.NoObf.InventoryItemRenderer.GetItemStackRenderInfo(_CeG6XgwLRetaWBDy7Vuk520qgAQ game, ItemSlot inSlot, EnumItemRenderTarget target)
   at Vintagestory.GameContent.EntityShapeRenderer.RenderHeldItem(Single dt, Boolean isShadowPass, Boolean right)
   at Vintagestory.GameContent.EntityShapeRenderer.DoRender3DOpaque(Single dt, Boolean isShadowPass)
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnRenderFrameShadows(Single dt)
   at _CpYdgbNZR5kyaDkgJT5mZbaEFXr._UsZi5ylBADLy2OH5Lo1MUzOq9qb(EnumRenderStage , Single )
   at _CeG6XgwLRetaWBDy7Vuk520qgAQ._UsZi5ylBADLy2OH5Lo1MUzOq9qb(EnumRenderStage , Single )
   at _CeG6XgwLRetaWBDy7Vuk520qgAQ._1nx5JhWedGp2NbeAYbMQcrPTaEJ(Single )
   at _CeG6XgwLRetaWBDy7Vuk520qgAQ._NsSwcIXfXJazD9UNZT7ucuORhzk(Single )
   at _WgVFI8rJI44rVQv521oBIjuWBxl._5QlzzE5xabSDGWATlodeOUeMK2b(Single )
   at _3bQQdrrneYMg7sUqAp4FsQ2XEfN._lShc2IGpVH0SOA7ydOiSsErkATH(Single )
   at _3bQQdrrneYMg7sUqAp4FsQ2XEfN._k3aLAIz3swGZO8aqd7pDrKnxALc(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _TPKNmFO93xHjVpa9EspHS9LXWzH._TsFumLr7TNG7iLJEc5uVYqFSeNG(_qRgBLoQrmfmbiUcRN39BpLhs3mN , String[] )
   at _QHQMScYkRrkBYdQDlv19fyrujIx._TsFumLr7TNG7iLJEc5uVYqFSeNG(ThreadStart )
-------------------------------

**Comment by Bloodsaw** [comment]
I can confirm that this error is reproducible:

**How to reproduce**

1. Fill a barrel with lime slack
2. duplicate the item with middle click
3. Place a new barrel
4. wait a few moments
5. game crashes

`Running on 64 bit Windows with 0 GB RAM 
Version: v1.12.9-rc.2 (Unstable)19.03.2020 20:01:09: Critical error occurred
System.IndexOutOfRangeException: Der Index war außerhalb des Arraybereichs.
   bei Vintagestory.GameContent.BlockBarrel.OnBeforeRender(ICoreClientAPI capi, ItemStack itemstack, EnumItemRenderTarget target, ItemRenderInfo& renderinfo)
   bei Vintagestory.Client.NoObf.InventoryItemRenderer.GetItemStackRenderInfo(_zHtHb9ee7akG2pQQxhr4D2rkwB game, ItemSlot inSlot, EnumItemRenderTarget target)
   bei Vintagestory.GameContent.EntityShapeRenderer.RenderHeldItem(Single dt, Boolean isShadowPass, Boolean right)
   bei Vintagestory.GameContent.EntityShapeRenderer.DoRender3DOpaque(Single dt, Boolean isShadowPass)
   bei Vintagestory.Client.NoObf.SystemRenderEntities.OnRenderFrameShadows(Single dt)
   bei _kcKrOSepVEkNdSScAPVPfaxh5aj._4ygjzkACoBBaWylqRAE6aCI1zdv(EnumRenderStage , Single )
   bei _zHtHb9ee7akG2pQQxhr4D2rkwB._4ygjzkACoBBaWylqRAE6aCI1zdv(EnumRenderStage , Single )
   bei _zHtHb9ee7akG2pQQxhr4D2rkwB._ubrLS78K4Ol3kuTTV4PienDcGnR(Single )
   bei _zHtHb9ee7akG2pQQxhr4D2rkwB._gAGmtjh81vboWc6VgDHFaoNhdLG(Single )
   bei _fSI1BMB65cwKGBNcX98G4kfwB7R._emHwz78R5gMeD1vHWpvZORwyJxk(Single )
   bei _tpQ1IERN2AMQ4hmdmlyG9BevCQP._NCuuF8mJPJiBaOfNuPzBvIB4fs1A(Single )
   bei _tpQ1IERN2AMQ4hmdmlyG9BevCQP._S14BA2XqynDvxHW2ozcDjnseezO(Single )
   bei Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   bei System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   bei OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   bei OpenTK.GameWindow.DispatchRenderFrame()
   bei OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   bei _kn7PO0uaevAeW8DtMdrhfUZoBG1._sTDuMc98lBVDhW4dFCrBjtOyF8CA(_RubDl5rVwhx3SBZRyR5HVRvNAUf , String[] )
   bei _gTATEydGikdlqm1ooAGFk9fvmWA._sTDuMc98lBVDhW4dFCrBjtOyF8CA(ThreadStart )`

**Comment by tyronx** [comment]
is this fixed already? because i cannot reproduce it with above steps

**Comment by Bloodsaw** [comment]
@tyronx you're right, it was already fixed with 1.12.11. Cannot reproduce this anymore.
